src/game/movement.rs

```diff
@@ -13,32 +13,31 @@ pub(super) fn plugin(app: &mut App) {
 
 /// Handle keyboard input to move the player.
 fn handle_player_movement_input(
-    keyboard_input: Res<ButtonInput<KeyCode>>,
-    mut query: Query<&mut Velocity, With<Player>>,
+    input: Res<ButtonInput<KeyCode>>,
+    mut player_velocity: Query<&mut Velocity, With<Player>>,
 ) {
     /// Since Bevy's default 2D camera setup is scaled such that
     /// one unit is one pixel, you can think of this as
     /// "How many pixels per second should the player move?"
     /// Note that physics engines may use different unit/pixel ratios.
     const SPEED: f32 = 240.0;
-    for mut velocity in query.iter_mut() {
-        velocity.0 = Vec3::ZERO;
 
-        if keyboard_input.pressed(KeyCode::KeyW) {
-            velocity.y += 1.0;
-        }
-        if keyboard_input.pressed(KeyCode::KeyS) {
-            velocity.y -= 1.0;
-        }
-        if keyboard_input.pressed(KeyCode::KeyA) {
-            velocity.x -= 1.0;
-        }
-        if keyboard_input.pressed(KeyCode::KeyD) {
-            velocity.x += 1.0;
-        }
-
-        // Need to normalize and scale because otherwise
-        // diagonal movement would be faster than horizontal or vertical movement.
-        velocity.0 = velocity.normalize_or_zero() * SPEED;
+    let mut intent = Vec3::ZERO;
```

Comment:
Not a fan of this rename. @MiniaczQ, opinion?

Reply:
Rationale: Input -> Intent -> Behavior opens the door for EnemyAi -> Intent -> Behavior, so this is the name I would use for the intermediate value.

Reply:
Although I suppose the intent wouldn't have speed applied.

Reply:
I'm ok with `intent` as long as it's axis-like, after we multiply it with SPEED it's very clearly `velocity` and no longer `intent`.